File: 136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#
from typing import List
# @lc code=start
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0

        for each in nums:
            result = result ^ each
            # XOR 비트 연산
        
        return result
    
# @lc code=end

for each in [[2,2,1], [4,1,2,1,2], [1]]:
    print(Solution.singleNumber(None, each))
    print()

File: 136.single-number.cpp
/*
 * @lc app=leetcode id=136 lang=cpp
 *
 * [136] Single Number
 */
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
 public:
  int singleNumber(vector<int>& nums) {
    int answer = 0;

    for (int i = 0; i < nums.size(); i++) {
      answer = nums[i] ^ answer;
    }

    return answer;
  }
};
// @lc code=end
